src/org/opendatakit/briefcase/ui/automation/AutomationPanel.java

```diff
@@ -2,51 +2,70 @@
 
 import static org.opendatakit.briefcase.util.FindDirectoryStructure.isWindows;
 
-import java.io.IOException;
-import java.nio.file.Files;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
 import java.nio.file.Path;
-import java.nio.file.Paths;
+import java.text.MessageFormat;
+import java.util.List;
 import javax.swing.JPanel;
 import org.bushe.swing.event.annotation.AnnotationProcessor;
 import org.opendatakit.briefcase.automation.AutomationConfiguration;
+import org.opendatakit.briefcase.model.BriefcaseFormDefinition;
+import org.opendatakit.briefcase.model.BriefcasePreferences;
 import org.opendatakit.briefcase.reused.BriefcaseException;
+import org.opendatakit.briefcase.util.FormCache;
 
 public class AutomationPanel {
   public static final String TAB_NAME = "Automation";
 
   private final AutomationPanelForm view;
-  private final AutomationConfiguration configuration;
+  private final FormCache formCache;
+  private final BriefcasePreferences appPreferences;
 
-
-  public AutomationPanel(AutomationPanelForm view, AutomationConfiguration configuration) {
+  public AutomationPanel(AutomationPanelForm view, BriefcasePreferences appPreferences, FormCache formCache) {
     AnnotationProcessor.process(this);
     this.view = view;
-    this.configuration = configuration;
-
-    view.onSelectScriptDir(path -> configuration.setScriptLocation(path));
+    this.formCache = formCache;
+    this.appPreferences = appPreferences;
 
-    view.onGenerate(() -> {
+    view.onGenerate(config -> {
+      System.out.println(config.getScriptLocation());
       if (isWindows())
-        generateScript("automation.bat", "echo %JAVA_HOME%");
+        generateScript("automation.bat",  config);
       else
-        generateScript("automation.sh", "echo $JAVA_HOME");
+        generateScript("automation.sh", config);
     });
   }
 
-  private void generateScript(String scriptName, String commands) {
+  private void generateScript(String scriptName, AutomationConfiguration configuration) {
     Path scriptDirPath = configuration.getScriptLocation().orElseThrow(() -> new BriefcaseException("No script location directory defined"));
-    Path scriptFilePath = Paths.get(scriptDirPath.toString() + scriptName);
+    String storageDir = appPreferences.getBriefcaseDir().orElseThrow(() -> new BriefcaseException("")).getParent().toString();
+    String template = "java -jar briefcase.jar --export --form_id {0} --storage_directory " +
+        storageDir + " --export_directory /tmp --export_filename {1}.csv";
     try {
-      Files.write(scriptFilePath, commands.getBytes());
-    } catch (IOException e) {
-      System.err.println(e.getMessage());
+      PrintWriter writer = new PrintWriter(scriptDirPath.toFile() + File.pathSeparator + scriptName);
```

Comment:
I think your NIO2 version didn't work because you weren't building the scriptFilePath correctly. You need to use the `Path.resolve()` method for that.

Reply:
Also, this version would weirdly use `:` as `File.pathSeparator` which produced a wrong path and made the whole this fail. With `Path.resolve()` you don't need to care about that.